sql/importer: TestCreateStatsAfterImport failed

sql/importer.TestCreateStatsAfterImport [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9287167?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9287167?buildTab=artifacts#/) on master @ [143b63a6a27aeb286637dd2e5abddafdf0c51874](https://github.com/cockroachdb/cockroach/commits/143b63a6a27aeb286637dd2e5abddafdf0c51874):


```
=== RUN   TestCreateStatsAfterImport
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestCreateStatsAfterImport6913716
    test_log_scope.go:79: use -show-logs to present logs inline
    sql_runner.go:103: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            				github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:6570
        	Error:      	Received unexpected error:
        	            	query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:314
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/importer.TestCreateStatsAfterImport
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:6570
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	  | 	  FROM [SHOW STATISTICS FOR TABLE t]': expected:
        	            	  | __auto__, {i}, 2, 2, 0
        	            	  | __auto__, {t}, 2, 2, 0
        	            	  | got:
        	            	  | __import__, {i}, 2, 0, 0
        	            	  | __import__, {t}, 2, 0, 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestCreateStatsAfterImport6913716
--- FAIL: TestCreateStatsAfterImport (54.57s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-sessions
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsAfterImport.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26079

sql/importer.TestCreateStatsAfterImport [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9377180?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9377180?buildTab=artifacts#/) on master @ [99102ddf4b7602788b422366f1acc14b81c64d24](https://github.com/cockroachdb/cockroach/commits/99102ddf4b7602788b422366f1acc14b81c64d24):


```
=== RUN   TestCreateStatsAfterImport
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestCreateStatsAfterImport1291642969
    test_log_scope.go:79: use -show-logs to present logs inline
    sql_runner.go:103: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            				github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:6596
        	Error:      	Received unexpected error:
        	            	query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:314
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:116
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:311
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/importer.TestCreateStatsAfterImport
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql/importer/import_stmt_test.go:6596
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1446
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1594
        	            	Wraps: (2) query 'SELECT statistics_name, column_names, row_count, distinct_count, null_count
        	            	  | 	  FROM [SHOW STATISTICS FOR TABLE t]': expected:
        	            	  | __auto__, {i}, 2, 2, 0
        	            	  | __auto__, {t}, 2, 2, 0
        	            	  | got:
        	            	  | __import__, {i}, 2, 0, 0
        	            	  | __import__, {t}, 2, 0, 0
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    panic.go:540: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/73dfce30b9a5630b1b4dabed3c94b32c/logTestCreateStatsAfterImport1291642969
--- FAIL: TestCreateStatsAfterImport (55.24s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCreateStatsAfterImport.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/importer/import_stmt_test.go
+++ b/pkg/sql/importer/import_stmt_test.go
@@ -2013,7 +2013,7 @@ func TestImportRowLimit(t *testing.T) {
 
 	t.Run("row limit multiple csv", func(t *testing.T) {
 		sqlDB.Exec(t, `CREATE DATABASE test; USE test`)
-		defer sqlDB.Exec(t, (`DROP DATABASE test`))
+		defer sqlDB.Exec(t, `DROP DATABASE test`)
 
 		data = "pear\navocado\nwatermelon\nsugar"
 		sqlDB.Exec(t, `CREATE TABLE t (s STRING)`)
@@ -6609,18 +6609,20 @@ func TestCreateStatsAfterImport(t *testing.T) {
 	sqlDB.Exec(t, "IMPORT PGDUMP ($1) WITH ignore_unsupported_statements", "nodelocal://1/cockroachdump/dump.sql")
 
 	// Verify that statistics have been created.
+	// Depending on timing, the statistics name may either be __auto__ or
+	// __import__, so we don't check the name in the results.
 	sqlDB.CheckQueryResultsRetry(t,
-		`SELECT statistics_name, column_names, row_count, distinct_count, null_count
+		`SELECT column_names, row_count, distinct_count, null_count
 	  FROM [SHOW STATISTICS FOR TABLE t]`,
 		[][]string{
-			{"__auto__", "{i}", "2", "2", "0"},
-			{"__auto__", "{t}", "2", "2", "0"},
+			{"{i}", "2", "2", "0"},
+			{"{t}", "2", "2", "0"},
 		})
 	sqlDB.CheckQueryResultsRetry(t,
-		`SELECT statistics_name, column_names, row_count, distinct_count, null_count
+		`SELECT column_names, row_count, distinct_count, null_count
 	  FROM [SHOW STATISTICS FOR TABLE a]`,
 		[][]string{
-			{"__auto__", "{i}", "1", "1", "0"},
+			{"{i}", "1", "1", "0"},
 		})
 }
 

